plugins/askrene/explain_failure.c

```diff
@@ -236,7 +236,8 @@ const char *explain_failure(const tal_t *ctx,
 	hops = route_from_dijkstra(tmpctx, rq->gossmap, dij, srcnode,
 				   AMOUNT_MSAT(0), 0);
 	if (!hops)
-		return tal_fmt(ctx, "There is no connection between source and destination at all");
+		return rq_log(ctx, rq, LOG_INFORM,
```

Comment:
I am not sure we should send logs from inside `explain_failure`. Maybe we should only use it
to get useful messages. Then the caller will make the logs.

On the other hand, if we are going to let `explain_failure` to write logs we might as well need to log the failure branches in lines 223 and 229.

Reply:
Yes, I started doing this when I was trying to reconstruct what happens from the debug logs: too much was missing.  It probably does need to go deeper, you're right!